xray/segment.go

```diff
@@ -240,6 +244,9 @@ func BeginSubsegment(ctx context.Context, name string) (context.Context, *Segmen
 	seg.StartTime = float64(time.Now().UnixNano()) / float64(time.Second)
 	seg.InProgress = true
 
+	// check whether segment is dummy or not based on sampling decision
+	seg.isDummy()
```

Comment:
can we fail faster by putting this immediately after `seg.ParentSegment = parent.ParentSegment`? We don't bother  keeping the footprint in the parent segment as we won't send it out anyway.

Reply:
Yeah. we can put this right after seg.ParentSegment = parent.ParentSegment.